File: padroesgof/src/main/java/com/padroesestruturais/Adapter/BrokerABCAdapter.java
package com.padroesestruturais.Adapter;

public class BrokerABCAdapter implements BrokerPagamento {
    
    @Override
    public void pagarCartaoCredito(CartaoCredito cartao, double valor) {

        ABCPagamentos brokerPagamento = new ABCPagamentos();

        brokerPagamento.pagarEmCartaoCredito(cartao.getNumero(), cartao.getNome(), cartao.getCvv(), cartao.getValidade(), valor);
    }
}

File: PadroesGofCSharp/Adapters/BrokerABCAdapter.cs
using PadroesGofCSharp.FormasDePagamento;
using PadroesGofCSharp.Models;

namespace PadroesGofCSharp.Adapters;

public class BrokerABCAdapter : BrokerPagamento
{
    public void pagarCartaoCredito(CartaoCredito cartao, decimal valor)
    {
        ABCPagamentos brokerPagamento = new ABCPagamentos();

        brokerPagamento.pagarEmCartaoCredito(cartao.getNumero(), cartao.getNome(), cartao.getValidade(), cartao.getCvv(), valor);
    }
}